bootcd/rpms/genesis_scripts/src/root-bash_profile

```diff
@@ -1,14 +1,25 @@
 
-GENESIS_MODE="$(grep -o -e 'GENESIS_MODE=[^ ]*' /proc/cmdline | cut -c14-)"
-
-# setup the ruby environment we need
-source /etc/profile.d/rvm.sh
-rvm list
-ruby --version
-
-if [[ ! -z $GENESIS_MODE ]] && [ ! -e "/var/run/bootloader-autorun.lock" ]
-  then
-    touch /var/run/bootloader-autorun.lock
-    # disable task execution prompting when doing initial target run
-    GENESIS_PROMPT_TIMEOUT=0  /usr/bin/genesis-bootloader 2>&1 | tee /var/log/genesis-bootloader.log
+# Genesis modifications
+
+# Get our configuration
+source /etc/sysconfig/genesis
+
+# Try for two seconds to find a pidfile
+for i in {1..20}
```

Comment:
This whole loop may be simplified with lsb init script functions from sourcing /lib/lsb/init-functions. There are helpers to read PIDs in a reliable manner